9.5.0-winxp toolchain profile will not build since libpthread was merged

Error in pass2:
```
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/./gcc/xgcc -shared-libgcc -B/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/./gcc -nostdinc++ -L/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/src -L/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/src/.libs -L/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/libsupc++/.libs -B/opt/toolchains/dc/sh-elf/sh-elf/bin/ -B/opt/toolchains/dc/sh-elf/sh-elf/lib/ -isystem /opt/toolchains/dc/sh-elf/sh-elf/include -isystem /opt/toolchains/dc/sh-elf/sh-elf/sys-include    -x c++-header -nostdinc++ -g -O2  -I/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/sh-elf -I/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include -I/opt/toolchains/dc/kos/utils/dc-chain/gcc-9.5.0/libstdc++-v3/libsupc++  -O2 -g /opt/toolchains/dc/kos/utils/dc-chain/gcc-9.5.0/libstdc++-v3/include/precompiled/stdc++.h -o sh-elf/bits/stdc++.h.gch/O2g.gch
1440.9 In file included from /opt/toolchains/dc/kos/utils/dc-chain/gcc-9.5.0/libstdc++-v3/include/precompiled/stdc++.h:122:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_rdlock(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:113:12: error: ‘pthread_rwlock_rdlock’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   113 |   { return pthread_rwlock_rdlock (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_tryrdlock(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:116:12: error: ‘pthread_rwlock_tryrdlock’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   116 |   { return pthread_rwlock_tryrdlock (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_wrlock(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:119:12: error: ‘pthread_rwlock_wrlock’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   119 |   { return pthread_rwlock_wrlock (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_trywrlock(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:122:12: error: ‘pthread_rwlock_trywrlock’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   122 |   { return pthread_rwlock_trywrlock (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_unlock(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:125:12: error: ‘pthread_rwlock_unlock’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   125 |   { return pthread_rwlock_unlock (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_destroy(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:128:12: error: ‘pthread_rwlock_destroy’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   128 |   { return pthread_rwlock_destroy (__rwlock); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_init(pthread_rwlock_t*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:131:12: error: ‘pthread_rwlock_init’ was not declared in this scope; did you mean ‘pthread_rwlock_t’?
1440.9   131 |   { return pthread_rwlock_init (__rwlock, NULL); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~
1440.9       |            pthread_rwlock_t
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_timedrdlock(pthread_rwlock_t*, const timespec*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:136:12: error: ‘pthread_rwlock_timedrdlock’ was not declared in this scope
1440.9   136 |   { return pthread_rwlock_timedrdlock (__rwlock, __ts); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex: In function ‘int std::__glibcxx_rwlock_timedwrlock(pthread_rwlock_t*, const timespec*)’:
1440.9 /opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include/shared_mutex:140:12: error: ‘pthread_rwlock_timedwrlock’ was not declared in this scope
1440.9   140 |   { return pthread_rwlock_timedwrlock (__rwlock, __ts); }
1440.9       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
1440.9 make[5]: *** [Makefile:1869: sh-elf/bits/stdc++.h.gch/O2g.gch] Error 1
1440.9 make[5]: Leaving directory '/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3/include'
1440.9 make[4]: *** [Makefile:562: all-recursive] Error 1
1440.9 make[4]: Leaving directory '/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3'
1440.9 make[3]: *** [Makefile:487: all] Error 2
1440.9 make[3]: Leaving directory '/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2/sh-elf/libstdc++-v3'
1440.9 make[2]: *** [Makefile:12162: all-target-libstdc++-v3] Error 2
1440.9 make[2]: Leaving directory '/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2'
1440.9 make[1]: *** [Makefile:957: all] Error 2
1440.9 make[1]: Leaving directory '/opt/toolchains/dc/kos/utils/dc-chain/build-gcc-sh-elf-9.5.0-pass2'
```

--- a/addons/libpthread/pthread-internal.h
+++ b/addons/libpthread/pthread-internal.h
@@ -7,6 +7,9 @@
 #ifndef __PTHREAD_INTERNAL_H
 #define __PTHREAD_INTERNAL_H
 
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
 #define __PTHREAD_HAVE_ATTR_TYPE        1
 #define __PTHREAD_HAVE_MUTEX_TYPE       1
 #define __PTHREAD_HAVE_COND_TYPE        1
@@ -80,5 +83,7 @@ STATIC_ASSERT(sizeof(pthread_barrier_t) == __PTHREAD_BARRIER_SIZE)
 STATIC_ASSERT(__PTHREAD_BARRIER_SIZE == THD_BARRIER_SIZE)
 
 #undef STATIC_ASSERT
+
+__END_DECLS
 
 #endif /* !__PTHREAD_INTERNAL_H */

--- a/include/sys/_pthreadtypes.h
+++ b/include/sys/_pthreadtypes.h
@@ -8,6 +8,9 @@
 #ifndef __SYS_PTHREADTYPES_H
 #define __SYS_PTHREADTYPES_H
 
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
 typedef unsigned long int pthread_t;
 
 typedef struct pthread_mutexattr_t {
@@ -99,5 +102,7 @@ typedef union pthread_barrier_t {
 
 #undef __PTHREAD_BARRIER_SIZE
 #endif /* !__PTHREAD_HAVE_BARRIER_TYPE */
+
+__END_DECLS
 
 #endif /* !__SYS_PTHREADTYPES_H */

--- a/include/sys/lock.h
+++ b/include/sys/lock.h
@@ -19,6 +19,9 @@
 #ifndef __SYS_LOCK_H__
 #define __SYS_LOCK_H__
 
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
 /** \cond */
 
 typedef struct {
@@ -62,5 +65,7 @@ int __newlib_lock_try_acquire_recursive(__newlib_recursive_lock_t*);
 void __newlib_lock_release_recursive(__newlib_recursive_lock_t*);
 
 /** \endcond */
+
+__END_DECLS
 
 #endif /* __SYS_LOCK_H__ */

--- a/include/sys/stdio.h
+++ b/include/sys/stdio.h
@@ -8,6 +8,9 @@
 #ifndef _NEWLIB_STDIO_H
 #define _NEWLIB_STDIO_H
 
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
 // Cribbed from newlib sys/stdio.h
 
 /* Internal locking macros, used to protect stdio functions.  In the
@@ -30,5 +33,7 @@
 #include <unistd.h>
 #include <stdarg.h>
 #include <kos/fs.h>
+
+__END_DECLS
 
 #endif /* _NEWLIB_STDIO_H */

